public/app/features/integrated-alerting/components/Table/Pagination/Pagination.test.tsx

```diff
@@ -0,0 +1,311 @@
+import React from 'react';
+import { shallow, ShallowWrapper } from 'enzyme';
+import { dataQa } from '@percona/platform-core';
+import { Pagination } from './Pagination';
+import { Messages } from './Pagination.messages';
+
+const simulateNextClick = (wrapper: ShallowWrapper, numberOfClicks: number) => {
+  for (let i = 0; i < numberOfClicks; i++) {
+    wrapper.find(dataQa('next-page-button')).simulate('click');
+  }
+};
+
+const simulatePreviousClick = (wrapper: ShallowWrapper, numberOfClicks: number) => {
+  for (let i = 0; i < numberOfClicks; i++) {
+    wrapper.find(dataQa('previous-page-button')).simulate('click');
+  }
+};
+
+describe('Pagination', () => {
+  it('should render at least one page', () => {
+    const wrapper = shallow(
+      <Pagination totalItems={0} pageCount={1} pageSizeOptions={[]} pageSize={3} nrRowsOnCurrentPage={0} />
+    );
+    expect(wrapper.find(dataQa('page-button')).length).toBe(1);
+    expect(wrapper.find(dataQa('page-button')).prop('variant')).toBe('primary');
+  });
+
+  it('should disable left navigation buttons when in first page', () => {
+    const wrapper = shallow(
+      <Pagination totalItems={30} pageCount={10} pageSizeOptions={[]} pageSize={3} nrRowsOnCurrentPage={3} />
+    );
+    expect(wrapper.find(dataQa('previous-page-button')).props().disabled).toBeTruthy();
+    expect(wrapper.find(dataQa('first-page-button')).props().disabled).toBeTruthy();
+  });
+
+  it('should disable right navigation buttons when in last page', () => {
+    const wrapper = shallow(
+      <Pagination totalItems={10} pageCount={1} pageSizeOptions={[]} pageSize={10} nrRowsOnCurrentPage={10} />
+    );
+    expect(wrapper.find(dataQa('next-page-button')).props().disabled).toBeTruthy();
+    expect(wrapper.find(dataQa('last-page-button')).props().disabled).toBeTruthy();
+  });
```

Comment:
I think these and other cases should not be testes, since they're part of the library and are not developed by us

Reply:
Right, but assuming the library's implementation is right, this certifies that `activePageIndex` matches our expectation, which is what we are testing (`disabled={activePageIndex === 0 / pageCount - 1 / other condition}`)

Reply:
If there's a branching you want to test in our code, then it's ok, but other than that our unit tests should be focus on our units of code : )

Reply:
👍